client/src/pages/forms/SignupForm.js

```diff
@@ -0,0 +1,162 @@
+import React from "react";
+import { useState, useEffect } from "react";
+
+import "./Form.css";
+
+
+function SignupForm() {
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	const initialDetails = {
+		Name: "",
+		Class: "",
+		Username: "",
+		Password: "",
+		ConfirmPassword: "",
+	};
+	const [details, setDetails] = useState(initialDetails);
+	const [errors, setErrors] = useState({});
+	const [submit, setSubmit] = useState(false);
+
+	function submitHandler(e) {
+		e.preventDefault();
+		setErrors(validate(details));
+		setSubmit(true);
+	}
+	useEffect(() => {
+		console.log("errors", errors);
+		if (Object.keys(errors).length === 0 && submit) {
+			setDetails(initialDetails);
+			console.log("details entered:", details);
+		}
+	}, [details, errors, initialDetails, submit]);
+
+	const validate = (details) => {
+		const errors = {};
+		const userNameRegex = /^[a-z0-9]+(?:[ _.-][a-z0-9]+)*$/;
+
+
+		if (!details.Name) {
+			errors.Name = "Name is required";
+		} else if (details.Name.length < 3) {
+			errors.Name = "Please enter your full name";
+		}
+		if (!details.Class) {
+			errors.Class = "Class is required";
+		} else if (details.Class.length < 3 || details.Class.length > 7) {
+			errors.Class = "Please enter the correct Class. i.e: WM3, NW4";
```

Comment:
Class will be a dropdown so you won't need to validate it, every value the user can pick will be guaranteed valid

Reply:
Done :)